core/tee/tee_enc_fs.c

```diff
@@ -0,0 +1,731 @@
+		goto exit;
+
+	if (fdp->flags & TEE_FS_O_RDONLY)
+		goto exit;
+
+	res = update_file_size(fdp, length);
+
+exit:
+	return res;
+}
+
+static int tee_fs_mkdir(const char *path, tee_fs_mode_t mode)
+{
+	int res = -1;
+	struct tee_fs_rpc head = { 0 };
+	uint32_t len;
+
+	len = strlen(path) + 1;
+	if (len > TEE_FS_NAME_MAX)
+		goto exit;
+
+	head.op = TEE_FS_MKDIR;
+	head.flags = mode;
+
+	res = tee_fs_send_cmd(&head, (void *)path, len, TEE_FS_MODE_IN);
+	if (!res)
+		res = head.res;
+
+exit:
+	return res;
+}
+
+static tee_fs_dir *tee_fs_opendir(const char *name)
+{
+	struct tee_fs_rpc head = { 0 };
+	uint32_t len;
+	struct tee_fs_dir *dir = NULL;
+
+	len = strlen(name) + 1;
+	if (len > TEE_FS_NAME_MAX)
+		goto exit;
+
+	head.op = TEE_FS_OPENDIR;
+
+	if (tee_fs_send_cmd(&head, (void *)name, len, TEE_FS_MODE_IN))
+		goto exit;
+
+	if (head.res < 0)
+		goto exit;
+
+	dir = malloc(sizeof(struct tee_fs_dir));
+	if (dir == NULL) {
+		int nw_dir = head.res;
+
+		memset(&head, 0, sizeof(head));
+		head.op = TEE_FS_CLOSEDIR;
+		head.arg = nw_dir;
+		tee_fs_send_cmd(&head, NULL, 0, TEE_FS_MODE_NONE);
+		goto exit;
+	}
+
+	dir->nw_dir = head.res;
+	dir->d.d_name = NULL;
+
+exit:
+	return dir;
+}
+
+static int tee_fs_closedir(tee_fs_dir *d)
+{
+	int res = -1;
+	struct tee_fs_rpc head = { 0 };
+
+	if (d == NULL) {
+		res = 0;
+		goto exit;
+	}
+
+	head.op = TEE_FS_CLOSEDIR;
+	head.arg = (int)d->nw_dir;
+
+	res = tee_fs_send_cmd(&head, NULL, 0, TEE_FS_MODE_NONE);
+	if (!res)
+		res = head.res;
+
+exit:
+	if (d)
+		free(d->d.d_name);
+	free(d);
+
+	return res;
+}
+
+static struct tee_fs_dirent *tee_fs_readdir(tee_fs_dir *d)
+{
+	struct tee_fs_dirent *res = NULL;
+	struct tee_fs_rpc head = { 0 };
+	char fname[TEE_FS_NAME_MAX + 1];
+
+	if (!d)
+		goto exit;
+
+	head.op = TEE_FS_READDIR;
+	head.arg = (int)d->nw_dir;
+
+	if (tee_fs_send_cmd(&head, fname, sizeof(fname), TEE_FS_MODE_OUT))
+		goto exit;
+
+	if (head.res < 0)
+		goto exit;
+
+	if (!head.len || head.len > sizeof(fname))
+		goto exit;
+
+	fname[head.len - 1] = '\0'; /* make sure it's zero terminated */
+	free(d->d.d_name);
+	d->d.d_name = strdup(fname);
+	if (!d->d.d_name)
+		goto exit;
+
+	res = &d->d;
+exit:
+	return res;
+}
+
+static int tee_fs_rmdir(const char *name)
+{
+	int res = -1;
+	struct tee_fs_rpc head = { 0 };
+	uint32_t len;
+
+	len = strlen(name) + 1;
+	if (len > TEE_FS_NAME_MAX)
+		goto exit;
+
+	head.op = TEE_FS_RMDIR;
+
+	res = tee_fs_send_cmd(&head, (void *)name, len, TEE_FS_MODE_IN);
+	if (!res)
+		res = head.res;
+
+exit:
+	return res;
+}
+
+static int tee_fs_access(const char *name, int mode)
+{
+	int res = -1;
+	struct tee_fs_rpc head = { 0 };
+	uint32_t len;
+
+	len = strlen(name) + 1;
+	if (len > TEE_FS_NAME_MAX)
+		goto exit;
+
+	head.op = TEE_FS_ACCESS;
+	head.flags = mode;
+
+	res = tee_fs_send_cmd(&head, (void *)name, len, TEE_FS_MODE_IN);
+	if (!res)
+		res = head.res;
+
+exit:
+	return res;
+}
+
+struct tee_file_operations tee_file_ops = {
+	.open = tee_fs_open,
```

Comment:
Could the functions `tee_fs_open`... be renamed in `tee_enc_fs_xxx` to avoid confusion with the standard fs function?